src/main.ts

```diff
@@ -101,15 +101,34 @@ async function run() {
       const linesChanged = +core.getInput("linesChanged");
       const filesChanged = +core.getInput("filesChanged");
 
-      if (diffContains || diffDoesNotContain || filesChanged || linesChanged) {
-        core.info("Checking diff contents");
-        const parsedDiff = await getDiff(octokit, repository, pull_request);
-        core.setOutput("numberOfFiles", parsedDiff.length);
-        if (filesChanged && parsedDiff.length != filesChanged) {
+      let filesChangedInPR: number;
+
+      // Check files changed first, before parsing diff
+      if (filesChanged) {
+        core.info("Checking number of files changed");
+        const owner = repository?.owner?.login;
+        const repo = repository?.name;
+        const pull_number = pull_request?.number;
+        
+        filesChangedInPR = await checkFilesChanged(
+          octokit,
+          owner,
+          repo,
+          pull_number
+        );
+        
+        if (filesChangedInPR != filesChanged) {
```

Comment:
fiiesChangedInPr should contain the files, not the number of files. Please change also this comparison.